File: Pointers/Pro8.cpp
// 8. Write a C program to search an x in array using pointers.

#include <stdio.h>

int searchElement(int *arr, int size, int x) {
    for (int i = 0; i < size; i++) {
        if (*(arr + i) == x)
            return i;
    }
    return -1;
}

int main() 
{
    int size, x, result;
    printf("Enter array size: ");
    scanf("%d", &size);
    int arr[size];
    printf("Enter array elements:\n");
    for (int i = 0; i < size; i++) {
        scanf("%d", &arr[i]);
    }
    printf("Enter the element to search: ");
    scanf("%d", &x);
    result = searchElement(arr, size, x);
    if (result != -1) {
        printf("Element %d found at index %d.\n", x, result);
    } 
	else {
        printf("Element %d not found in the array.\n", x);
    }
    return 0;
}

File: Assignment - 3_2nd Sem/Pro8.py
//8. Write a program to input a number and search in a list using binary search.

def binary_search(arr, target):
    low, high = 0, len(arr) - 1
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1
    return -1
arr = sorted([int(x) for x in input("Enter a sorted list of numbers: ").split()])
target = int(input("Enter the number to search: "))
index = binary_search(arr, target)
print(f"Number found at index {index}" if index != -1 else "Number not found.")
